### ISSUE TITLE
some issues with -v 0.3.10

### ISSUE THREAD

**Issue by bivol** [freedomofpress/securedrop#1511]
Hello, guys,

First of all - thanks for your hard work. SecureDrop is very useful tool for our work. We were getting a lot of leaks with the earlier versions and we are now setuping new servers.

I'm having some problems with the ansible playbook. I've commented some of the tasks with hope that I can proceed, but I finally gave up. I'm attaching some of the errors I encountered. I would love if somebody can help me in the process... I might be wrong somewhere from the beginning...

###### paxctl -Cpm grub_pax

I saw that there is already opened issue for that, but I wanted to write it here, as this is the first problem that a new installation would face.

> failed: [mon] (item=grub_pax) => {"changed": true, "cmd": ["paxctl", "-Cpm", "grub_pax"], "delta": "0:00:00.001722", "end": > "2016-12-28 18:23:28.021067", "failed": true, "item": "grub_pax", "rc": 1, "start": "2016-12-28 18:23:28.019345", "stderr": > "grub_pax: No such file or directory", "stdout": "", "stdout_lines": [], "warnings": []}
> failed: [app] (item=grub_pax) => {"changed": true, "cmd": ["paxctl", "-Cpm", "grub_pax"], "delta": "0:00:00.001910", "end": > "2016-12-28 18:23:28.097517", "failed": true, "item": "grub_pax", "rc": 1, "start": "2016-12-28 18:23:28.095607", "stderr": > "grub_pax: No such file or directory", "stdout": "", "stdout_lines": [], "warnings": []}

###### the field 'args' has an invalid value

> fatal: [mon]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'unicode object' has no attribute 'name'\n\nThe error appears to have been in '/home/amnesia/Tools/securedrop/install_files/ansible-base/roles/common/tasks/sysctl.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- sysctl: name=\"{{ item.name }}\" value=\"{{ item.value }}\" sysctl_set=yes state=present reload=yes\n  ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes.  Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}
> fatal: [app]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'unicode object' has no attribute 'name'\n\nThe error appears to have been in '/home/amnesia/Tools/securedrop/install_files/ansible-base/roles/common/tasks/sysctl.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- sysctl: name=\"{{ item.name }}\" value=\"{{ item.value }}\" sysctl_set=yes state=present reload=yes\n  ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes.  Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}

> fatal: [mon]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'unicode object' has no attribute 'name'\n\nThe error appears to have been in '/home/amnesia/Tools/securedrop/install_files/ansible-base/roles/common/tasks/apply_grsec_lock.yml': line 33, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- sysctl: name=\"{{ item.name }}\" value=\"{{ item.value }}\" sysctl_set=yes state=present reload=yes\n  ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes.  Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}
> fatal: [app]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'unicode object' has no attribute 'name'\n\nThe error appears to have been in '/home/amnesia/Tools/securedrop/install_files/ansible-base/roles/common/tasks/apply_grsec_lock.yml': line 33, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- sysctl: name=\"{{ item.name }}\" value=\"{{ item.value }}\" sysctl_set=yes state=present reload=yes\n  ^ here\nWe could be wrong, but this one looks like it might be an issue with\nmissing quotes.  Always quote template expression brackets when they\nstart a value. For instance:\n\n    with_items:\n      - {{ foo }}\n\nShould be written as:\n\n    with_items:\n      - \"{{ foo }}\"\n"}

> fatal: [mon]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'unicode object' has no attribute 'ip'\n\nThe error appears to have been in '/home/amnesia/Tools/securedrop/install_files/ansible-base/roles/common-mon/tasks/set_etc_hosts_for_prod_env.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- name: ensure /etc/hosts file has entry for app and monitor servers\n  ^ here\n"}

> fatal: [app]: FAILED! => {"failed": true, "msg": "the field 'args' has an invalid value, which appears to include a variable that is undefined. The error was: 'unicode object' has no attribute 'ip'\n\nThe error appears to have been in '/home/amnesia/Tools/securedrop/install_files/ansible-base/roles/common-app/tasks/set_etc_hosts_for_prod_env.yml': line 2, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n---\n- name: ensure /etc/hosts file has entry for app and monitor servers\n  ^ here\n"}

######  ossec_postfix_dependencies

> failed: [mon] (item=[u'ossec_postfix_dependencies']) => {"failed": true, "item": ["ossec_postfix_dependencies"], "msg": "No package matching 'ossec_postfix_dependencies' is available"}

######  mon_tor_instances

> failed: [mon] (item=mon_tor_instances) => {"elapsed": 300, "failed": true, "item": "mon_tor_instances", "msg": "Timeout when waiting for file /var/lib/tor/services/mon_tor_instances/hostname"}

######  No package matching 'app_deps' is available
> TASK [app : ensure secureDrop web application apt packages are installed] ******
> failed: [app] (item=[u'app_deps']) => {"failed": true, "item": ["app_deps"], "msg": "No package matching 'app_deps' is available"}

######  No package matching 'apache_packages' is available

> failed: [app] (item=[u'apache_packages']) => {"failed": true, "item": ["apache_packages"], "msg": "No package matching 'apache_packages' is available"}

######  Unable to find 'apache_files' in expected paths.
> failed: [app] (item=apache_files) => {"failed": true, "item": "apache_files", "msg": "Unable to find 'apache_files' in expected paths."}

Thanks for all your help! Please note that those might not be real issues, might be my mistakes. Thanks for you support.

**Comment by conorsch** [comment]
@bivol Sorry for the slow reply here. The issue you describe is without a doubt caused by a too-recent [sic] version of Ansible. Make sure you're installing the version of Ansible included in the Tails repos, i.e. 1.7.2, and the playbook should run just fine. See #1146 for discussion of Ansible versions—we may just start telling Admins to install the latest Ansible via pip and supporting that.

**Comment by bivol** [comment]
Hey, @conorsch,

Sorry to bother you again. Not going to open new ticket, but I'm now on Tails, and I bumped in new error:

```
fatal: [app] => error while evaluating conditional: not install local_packages
fatal: [mon] => error while evaluating conditional: not install local_packages
```

Where I can chat with some developers? We want to get this SecureDrop running as soon as it possible. We've created account on http://support.freedom.press, but seems it need to be validated by Administrator.

Thanks in advance for the reply.
